typescript/utils/src/__tests__/sleep.test.ts

```diff
@@ -0,0 +1,14 @@
+import { sleep } from "../sleep";
+
+jest.useFakeTimers();
+
+describe("sleep", () => {
+  it("waits for the specified amount of time", async () => {
+    const resolve = jest.fn();
+    const reject = jest.fn();
+    sleep(100).then(resolve).catch(reject);
+    jest.advanceTimersByTime(100);
```

Comment:
```suggestion
    jest.advanceTimersByTime(50);
    expect(resolve).not.toHaveBeenCalled();
    jest.advanceTimersByTime(60);
    expect(resolve).toHaveBeenCalled();
```

Reply:
it seems that my suggestion doesn't work as expected after all, but this might help: https://stackoverflow.com/questions/66391541/jest-advancetimersbytime-doesnt-work-when-i-try-to-test-my-retry-util-function/66415212#66415212

Reply:
It doesn't work as-is because you've set the second value to 60 whereas the sleep is 100 😉

Reply:
And because it's placed above the `await Promise.resolve()` line which forces the promise engine to have a new tick and resolve the Promise returned by sleep (can you feel the headache I had while implementing this test now? 😝)